### PR TITLE
feat(servicestage): add new application resource support

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Support a new ServiceStage resource and the configuration of environment variables.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
reference: #2030 

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support a new servicestage resource.
2. support the configutaion of environment variables.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
Pending
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/servicestage' TESTARGS='-run=TestAccApplication_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/servicestage -v -run=TestAccApplication_basic -timeout 360m -parallel 4
=== RUN   TestAccApplication_basic
=== PAUSE TestAccApplication_basic
=== CONT  TestAccApplication_basic
--- PASS: TestAccApplication_basic (250.83s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/servicestage      250.919s
```

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+	err = createEnvironmentVariables(client, d.Id(), d.Get("environment").(*schema.Set))
+	if err != nil {
+		return diag.Errorf("error creating environment variable: %s", err)
+	}
+
+	return resourceApplicationRead(ctx, d, meta)
+}
+
+func flattenEnvironmentVariables(d *schema.ResourceData, configs []applications.ConfigResp) []map[string]interface{} {
+	// Store all variables of the same environment in the corresponding list, and generate a map with the environment ID
```
Comment (hmbig2) [inline huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go]: s/d *schema.ResourceData/d.id()

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+	}
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("name", application.Name),
+		d.Set("description", application.Description),
+		d.Set("enterprise_project_id", application.EnterpriseProjectId),
+		d.Set("component_ids", flattenComponentIds(componentList)),
+	)
+	configs, err := applications.ListConfig(client, d.Id(), applications.ListConfigOpts{})
```
Comment (hmbig2) [inline huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go]: what is the differet between component and env config

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/servicestage' TESTARGS='-run=TestAccApplication_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/servicestage -v -run=TestAccApplication_basic -timeout 360m -parallel 4
=== RUN   TestAccApplication_basic
=== PAUSE TestAccApplication_basic
=== CONT  TestAccApplication_basic
--- PASS: TestAccApplication_basic (233.41s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/servicestage      234.178s
```

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go]: done.

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go]: The environment variable logic is written later than the main logic.

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+					"name":  variable.Name,
+					"value": variable.Value,
+				}
+			}
+
+			result = append(result, map[string]interface{}{
+				"id":       v.EnvironmentId,
+				"variable": variables,
+			})
+			break
```
Comment (ShiChangkuo) [inline huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go]: seems that **break** is unnecessary

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+	return resourceApplicationRead(ctx, d, meta)
+}
+
+func flattenEnvironmentVariables(appId string, configs []applications.ConfigResp) []map[string]interface{} {
+	// Store all variables of the same environment in the corresponding list, and generate a map with the environment ID
+	// as the key name.
+	result := make([]map[string]interface{}, 0)
+	for _, v := range configs {
+		if v.ApplicationId == appId && len(v.Configuration.EnvVariables) >= 1 {
```
Comment (ShiChangkuo) [inline huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go]: `v.ApplicationId == appId` is always true?

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+			},
+		},
+	}
+}
+
+func createEnvironmentVariables(client *golangsdk.ServiceClient, appId string, envSet *schema.Set) error {
+	for _, env := range envSet.List() {
+		envMap := env.(map[string]interface{})
+		variables := make([]applications.Variable, envMap["variable"].(*schema.Set).Len())
+		for i, v := range envMap["variable"].(*schema.Set).List() {
```
Comment (ShiChangkuo) [inline huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go]: suggest to use a variable to save `envMap["variable"].(*schema.Set)`

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go]: done.

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go]: yeah, and I think the length check is unnecessary too, so I removed both.

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/servicestage/resource_huaweicloud_servicestage_application.go]: done.

**Pull request merged**